src/main/java/com/tira/restaurants/validator/EmailValidator.java

```diff
@@ -0,0 +1,29 @@
+package com.tira.restaurants.validator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+
+public class EmailValidator implements ConstraintValidator<ValidEmail, String> {
+    private Pattern pattern;
+    private Matcher matcher;
+    private static final String EMAIL_PATTERN = "^[_A-Za-z0-9-\\+]+(\\.[_A-Za-z0-9-]+)*@" + "[A-Za-z0-9-]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,})$";
```

Comment:
You'd generally want to avoid writing your own email validation regexp, and try to use the ones already available in common libraries